src/core/thread/topology.hpp

```diff
@@ -831,10 +831,7 @@ class Child : public Neighbor,
     class AddressIteratorBuilder;
 
 public:
-    enum
-    {
-        kMaxRequestTlvs = 5,
-    };
+    static constexpr uint8_t kMaxRequestTlvs = 5;
```

Comment:
nit: maybe we can have a list of allowed tlv types.
```suggestion
      static constexpr Mle::Tlv::Type kRequestTlvsAllowed[] = {
          Mle::Tlv::kNetworkData,    Mle::Tlv::kRoute,   Mle::Tlv::kActiveDataset,
          Mle::Tlv::kPendingDataset, Mle::Tlv::kTimeout,
      };
      static constexpr uint8_t kMaxRequestTlvs = sizeof(kRequestTlvsAllowed) / sizeof(Mle::Tlv::Type);
```

Reply:
Thanks. It is a good suggestion but I would say we add it in a different PR (want to keep the scope of this PR to just constant `enum` changes and no other code/logic change). 

Also I'd say may be better to define this array (and then `kMaxRequestTlvs`)  in `Mle` module instead of in `Child` (and `topology.hpp`). Feels more natural for MLE module to define the array of allowed TLVs.